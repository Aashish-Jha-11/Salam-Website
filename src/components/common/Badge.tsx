import { cva, VariantProps } from "class-variance-authority";
import React, { FC, HTMLAttributes } from "react";

const span = cva("span", {
  variants: {
    intent: {
      primary: ["bg-light-primary", "text-primary"],
      success: ["bg-light-success", "text-success"],
      warning: ["bg-light-warning", "text-warning"],
      danger: ["bg-light-danger", "text-danger"],
    },
    size: {
      medium: [
        "py-1 sm:py-2",
        "px-2 sm:px-4",
        "font-Estedad-Medium",
        "text-[10px] sm:text-xs",
      ],
    },
    rounded: {
      full: ["rounded-full"],
    },
  },
  compoundVariants: [{ intent: "primary", size: "medium", rounded: "full" }],
  defaultVariants: { intent: "primary", size: "medium", rounded: "full" },
});

interface SpanProps
  extends HTMLAttributes<HTMLSpanElement>,
    VariantProps<typeof span> {}

export const Badge: FC<SpanProps> = ({
  className,
  intent,
  size,
  rounded,
  ...props
}) => (
  <span className={span({ intent, size, rounded, className })} {...props} />
);
