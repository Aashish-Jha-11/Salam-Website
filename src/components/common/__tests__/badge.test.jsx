'use strict';

import React from 'react';
import renderer from 'react-test-renderer';
import Badge from '../Badge';

it('renders correctly', () => {
  const tree = renderer
    .create('<Badge className="primary">آنلاین کار کن</Badge>').toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly', () => {
  const tree = renderer
    .create('<Badge className="primary medium full">آنلاین کار کن</Badge>').toJSON();
  expect(tree).toMatchSnapshot();
});
