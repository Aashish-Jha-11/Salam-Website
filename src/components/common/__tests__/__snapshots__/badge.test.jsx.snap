// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `"<Badge className="primary">آنلاین کار کن</Badge>"`;

exports[`renders correctly 2`] = `"<Badge className="primary medium full">آنلاین کار کن</Badge>"`;
